pdf-rendering-template/app/controllers/pdf_rendering_example_controller.rb

```diff
@@ -0,0 +1,49 @@
+class PdfRenderingExampleController < ApplicationController
+  def index
+  end
+
+  ##
+  # GET /pdf_rendering_example/:id
+  # * Renders the HTML version
+  #
+  # GET /pdf_rendering_example/:id.pdf
+  # * Generates the HTML version and converts it to PDF
+  #
+  #
+  def show
+    @example_num = params[:id]
+
+    respond_to do |format|
+      format.html do
+        render :show
+      end
+
+      format.pdf do
+        html = render_as_html(template_path: "pdf_rendering_example/show.html.erb",
+                              assigments: { example_num: @example_num })
+        pdf = generate_pdf_from(html: html)
+        send_file(pdf.file_path, filename: pdf.download_as_filename, type: pdf.mime_type)
+      end
+    end
+  end
+
+  private
+
+  def render_as_html(template_path:, assigments:)
+    ##
+    # Create a new instance of of ActionController::Renderer, passing it our
+    # current request environment to use as it's environment. This ensures that
+    # all the methods that would normally work in the view templates will
+    # continue to work e.g. `current_user` needs Warden in the environment.
+    #
+    # Details:
+    #   https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/renderer.rb
+    #
+    renderer = ApplicationController.renderer.new(request.env)
```

Comment:
Passing `request.env` ensures that `current_user` etc. work in the view templates. I found a number of blog posts which used `renderer.instance_variable_set(:@env, {...})` to set just the bits of the env that devise needs but, after a bit of source diving, passing the whole `request.env` seems like the cleanest and most reliable method.